src/Nerdbank.MessagePack/Converters/ArrayConverter`1.cs

```diff
@@ -152,4 +180,18 @@ public override async ValueTask WriteAsync(MessagePackAsyncWriter writer, TEleme
 			["items"] = context.GetJsonSchema(((IEnumerableTypeShape<TElement[], TElement>)typeShape).ElementType),
 		};
 	}
+
+	/// <summary>
+	/// Determines if a thrown exception should be wrapped with contextual information.
+	/// </summary>
+	/// <param name="ex">The thrown exception.</param>
+	/// <param name="cancellationToken">The cancellation token.</param>
+	/// <returns><see langword="true" /> if the exception should be wrapped; <see langword="false" /> otherwise.</returns>
+	/// <remarks>
+	/// We wrap all exceptions <em>except</em> <see cref="OperationCanceledException"/> if the cancellation token is cancelled.
+	/// In other words, the only time we allow any exception to escape is when the operation was cancelled, because
+	/// that is the intended behavior of cancellation tokens.
+	/// </remarks>
+	private static bool ShouldWrapSerializationException(Exception ex, CancellationToken cancellationToken)
```

Comment:
Instead of defining this in each converter class, declare it as `private protected` in the `MessagePackConverter<T>` base class.